web-client/src/components/types.ts

```diff
@@ -0,0 +1,20 @@
+import {Rule} from "../hooks/useRbac";
+import {AggregatedRoleBinding} from "../services/role";
+
+export interface RuleSet {
+  
+  readonly template?: string
+  readonly rules: Rule[],
+  /**
+   * ALL_NAMESPACES (from TemplatePair) is converted to ['all']
+   * @see AggregatedRoleBinding
+   */
+  readonly namespaces: string[]
+}
+
+/**
+ * NONE = no cluster wide access
+ * READ = can read all resources
+ * WRITE = can read/write all resources
+ */
+export type ClusterAccess = "none" | "read" | "write" | null
```

Comment:
Are sure about the `null`? Maybe this is not the case but most of the time it's better to make the type of the receiving variable optional (ex. `clusterAccess?: ClusterAccess` which is equivalent to `clusterAccess: ClusterAccess | null`) instead of making the type nullable

Reply:
Hello, you are of course correct, but I just ended up recreating what I've seen in the code logic. The variable can  exist but its set with null value (I interpret the '?' operator as undefined variable, javascript magic).

I think that we can handle that in a future refactor and remove both the undefined possibility and the null one because they can be both expressed with 'none' case as far I understand